fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registry/FlammableBlockRegistryImpl.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.content.registry;
+
+import com.google.common.base.Preconditions;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FireBlock;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.fabric.api.content.registry.v1.FlammableBlockRegistry;
+import net.fabricmc.fabric.mixin.content.registry.FireBlockAccessor;
+
+public class FlammableBlockRegistryImpl extends ContentRegistryImpl<Block, FlammableBlockRegistry.Entry> implements FlammableBlockRegistry {
+	public static final FlammableBlockRegistry FIRE_INSTANCE = new FlammableBlockRegistryImpl(Blocks.FIRE, "fire");
+
+	public FlammableBlockRegistryImpl(Block fireBlock, String prefix) {
+		super(prefix + "_flammable_block_registry", (block, entry) -> ((FireBlock) fireBlock).registerFlammableBlock(block, entry.getBurnChance(), entry.getSpreadChance()), block -> ((FireBlock) fireBlock).registerFlammableBlock(block, 0, 0), block -> new Entry(((FireBlockAccessor) fireBlock).getBurnChances().getInt(block), ((FireBlockAccessor) fireBlock).getSpreadChances().getInt(block)));
+		Preconditions.checkArgument(fireBlock instanceof FireBlock, "Block " + Registry.BLOCK.getId(fireBlock) + " is not a FireBlock and a FlammablcBlockRegistry cannot be created for it");
```

Comment:
Atleast split this over mulitple lines, its impossible to know whats going on here. Making the constutor take a `FireBlock` would remove a load of casts from this.

I assume this has been done this way to handle the 1.16 fire. You do the Preconditions.checkArgument after all the casts have failed so thats poinless as it is now anyway.

Reply:
Is it pointless? Those lambdas shouldn't be called until after the constructor

Also it's not that way to handle 1.16 fire (which doesn't spread so it wouldn't use this anyway), that's just how FireBlock has it implemented, it has a map per FireBlock (which is just 1 lol). I guess for once mojang actually planned on there being multiple types of fire or something.

The reason I didn't have it take a FireBlock is because the fields in `Blocks` such as `Blocks.FIRE` are stored as just the type `Block`, so requiring a FireBlock in the impl would require casting on the API user's side which seemed less clean to me.